test/Api.IntegrationTest/SecretsManager/Controllers/SecretsControllerTest.cs

```diff
@@ -145,28 +294,52 @@ public async Task UpdateSecret()
         AssertHelper.AssertRecent(updatedSecret.RevisionDate);
         AssertHelper.AssertRecent(updatedSecret.CreationDate);
         Assert.Null(updatedSecret.DeletedDate);
-        Assert.NotEqual(initialSecret.Value, updatedSecret.Value);
-        Assert.NotEqual(initialSecret.RevisionDate, updatedSecret.RevisionDate);
+        Assert.NotEqual(secret.Value, updatedSecret.Value);
+        Assert.NotEqual(secret.RevisionDate, updatedSecret.RevisionDate);
+    }
+
+    [Theory]
+    [InlineData(false, false)]
+    [InlineData(true, false)]
+    [InlineData(false, true)]
+    public async Task Delete_SmNotEnabled_NotFound(bool useSecrets, bool accessSecrets)
+    {
+        var (org, _) = await _organizationHelper.Initialize(useSecrets, accessSecrets);
+        await LoginAsync(_email);
+
+        var secret = await _secretRepository.CreateAsync(new Secret
+        {
+            OrganizationId = org.Id,
+            Key = _mockEncryptedString,
+            Value = _mockEncryptedString,
+            Note = _mockEncryptedString
+        });
+        var secretIds = new[] { secret.Id };
+
+        var response = await _client.PostAsJsonAsync("/secrets/delete", secretIds);
+        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
     }
 
     [Fact]
-    public async Task DeleteSecrets()
+    public async Task Delete_Owner_Success()
     {
-        var secretsToDelete = 3;
+        var (org, _) = await _organizationHelper.Initialize(true, true);
+        await LoginAsync(_email);
+
         var secretIds = new List<Guid>();
-        for (var i = 0; i < secretsToDelete; i++)
+        for (var i = 0; i < 3; i++)
         {
             var secret = await _secretRepository.CreateAsync(new Secret
             {
-                OrganizationId = _organization.Id,
+                OrganizationId = org.Id,
                 Key = _mockEncryptedString,
                 Value = _mockEncryptedString,
                 Note = _mockEncryptedString
             });
             secretIds.Add(secret.Id);
         }
 
-        var response = await _client.PostAsync("/secrets/delete", JsonContent.Create(secretIds));
+        var response = await _client.PostAsJsonAsync("/secrets/delete", secretIds);
```

Comment:
I did change some endpoints in my PR, so I'm sure there will be conflicts on one or two of these tests when I try and merge it into my branch 🤔